seqr/views/apis/project_api_tests.py

```diff
@@ -166,3 +166,54 @@ def test_empty_project_page_data(self):
         self.assertDictEqual(response_json['genesById'], {})
         self.assertDictEqual(response_json['mmeSubmissionsByGuid'], {})
         self.assertDictEqual(response_json['locusListsByGuid'], {})
+
+
+# Tests for AnVIL access disabled
+class LocalProjectAPITest(AuthenticationTestCase, ProjectAPITest):
+    fixtures = ['users', '1kg_project', 'reference_data']
+
+
+# Test for permissions from AnVIL only
+class AnvilProjectAPITest(AnvilAuthenticationTestCase, ProjectAPITest):
+    fixtures = ['users', 'social_auth', '1kg_project', 'reference_data']
+
+    def test_create_update_and_delete_project(self):
+        super(AnvilProjectAPITest, self).test_create_update_and_delete_project()
+        self.mock_list_workspaces.assert_not_called()
+        self.mock_service_account.get.assert_not_called()
+
+    def test_project_page_data(self):
+        super(AnvilProjectAPITest, self).test_project_page_data()
+        self.mock_list_workspaces.assert_not_called()
+        self.mock_service_account.get.assert_called_with(
+            'api/workspaces/my-seqr-billing/anvil-1kg project n\u00e5me with uni\u00e7\u00f8de/acl')
+        self.assertEqual(self.mock_service_account.get.call_count, 5)
```

Comment:
Can you update this test to test what the content of the project `collaborators` key in the response data? I would expect it to be different in the anvil and non-anvil cases so I would like that to be tested. So after line 94 above you could add something like 
```
collaborators = response_json['projectsByGuid'][PROJECT_GUID]['collaborators']
self.assertEqual(len(collaborators), NUM_COLLABORATORS)
self.assertEqual(len([c for c in collaborators if c['isAnvil']), NUM_ANVIL_COLLABORATORS)
```
And then for each of the local/anvil/mix cases you could define the appropriate variables

Reply:
Good point. But I do it differently because all the collaborators have 'isAnvil' set when the Terra API is enabled. I compare the collaborator list against the expected list.